include/charstr.h

```diff
@@ -62,6 +62,25 @@ char charstr_ucase_char(char c);
 /* The return value is -1U if c is not a hexadecimal digit. */
 unsigned charstr_digit_value(char c);
 
+/* charstr_to_unsigned() and charstr_to_integer() parse a buffer of
+ * characters into a 64-bit unsigned or signed integer. The functions
+ * are similar to strtol(3) (qv), but:
+ *  - The locale is ignored.
+ *  - errno is set to 0 if no error is encountered.
+ *  - base must be 0 or between 2 and 16, or errno is set to EINVAL.
+ *  - Initial whitespace is not skipped.
+ *  - If endptr and *endptr are non-NULL on input, *endptr is used to
+ *    limit the sequence.
+ *  - In case of an overflow, the returned value wraps naturally, but
+ *    errno is set to ERANGE
+ *  - If the buffer does not contain a valid unsigned integer, errno
+ *    is set to EILSEQ.
+ *  - In case of an error return, the value of *end is undefined. */
+uint64_t charstr_to_unsigned(const char *buffer, const char **end,
```

Comment:
Is there a particular reason to mirror the `strtol` interface? Looking at `float.h` again, we have
```
ssize_t binary64_from_string(const char *buffer, size_t size, uint64_t *value);
```
which I find better. What about providing a similar interface for all string to number conversions in a new module, say `charconv.h` (inspired by C++ https://en.cppreference.com/w/cpp/header/charconv)?